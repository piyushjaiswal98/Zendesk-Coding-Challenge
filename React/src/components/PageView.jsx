import React, { useEffect, useState } from 'react';
import './PageView.scss';
import TicketList from '../components/TicketList';
import ReactPaginate from 'react-paginate';

const Pageview = (props) => {

    const [data, setData] = useState([]);
    const [cdata, setCdata] = useState([]);

    const PER_PAGE = 25;
    const pageCount = Math.ceil(data.length / PER_PAGE);

    useEffect(() => {
        // console.log(props.data);
        setData(props.data);
        if(props.data && props.data.length>25)
        {
        var d = props.data
                .slice(0, PER_PAGE);
        setCdata(d); }
        else{
            setCdata(props.data);
        }
    },[props.data]);

    function handlePageClick({ selected: selectedPage }) {
        var d = data
                .slice(selectedPage*PER_PAGE, (selectedPage*PER_PAGE) + PER_PAGE);
        setCdata(d);
    }

    return(

        <div className="pageview">
                        
            <TicketList ticketList={cdata}/>
                        {/* <br/> */}
            <div className="page-class">
            <ReactPaginate
                            previousLabel={"← Previous"}
                            nextLabel={"Next →"}
                            pageCount={pageCount}
                            onPageChange={handlePageClick}
                            pageClassName="page-item"
                            pageLinkClassName="page-link"
                            previousClassName="page-item"
                            previousLinkClassName="page-link"
                            nextClassName="page-item"
                            nextLinkClassName="page-link"
                            breakLabel="..."
                            breakClassName="page-item"
                            breakLinkClassName="page-link"
                            containerClassName="pagination"
                            activeClassName="active"
                            renderOnZeroPageCount={null}        
            /> 
            </div>

        </div>
    )

}

export default Pageview;

