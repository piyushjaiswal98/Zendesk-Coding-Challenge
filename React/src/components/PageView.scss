@import "https://cdn.jsdelivr.net/npm/bootstrap@5.1.3/dist/css/bootstrap.min.css";

.pageview{
    width: 100%;
    height: 100%;
    .page-class{
        margin-top: 15px;
        margin-left: 20%;
    }
}