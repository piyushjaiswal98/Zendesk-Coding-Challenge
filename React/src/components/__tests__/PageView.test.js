import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import PageView from '../PageView';

afterEach(()=> {
    cleanup();
})

const data = [{ 
    "ticket": { 
        "url": "https://zccstudents3001.zendesk.com/api/v2/tickets/4.json", 
        "id": 4, 
        "created_at": "2021-11-24T21:20:18Z", 
        "updated_at": "2021-11-24T21:20:18Z", 
        "subject": "ad sunt qui aute ullamco", 
        "raw_subject": "ad sunt qui aute ullamco", 
        "description": "Sunt incididunt officia proident elit anim ullamco reprehenderit ut. Aliqua sint amet aliquip cillum minim magna consequat excepteur fugiat exercitation est exercitation. Adipisicing occaecat nisi aliqua exercitation.\n\nAute Lorem aute tempor sunt mollit dolor in consequat non cillum irure reprehenderit. Nulla deserunt qui aliquip officia duis incididunt et est velit nulla irure in fugiat in. Deserunt proident est in dolore culpa mollit exercitation ea anim consequat incididunt. Mollit et occaecat ullamco ut id incididunt laboris occaecat qui.", 
        "status": "open", 
    } 
}];

test('Should render PageView Component', () => {
    render(<PageView data={data}/>);
    const PageViewElement = screen.getByTestId('PageView-1');
    expect(PageViewElement).toBeInTheDocument();
    expect(PageViewElement).toHaveClass("pageview");
})

test('Matches Snapshot', () => {
    const tree = renderer.create(<PageView data={data}/>).toJSON();
    expect(tree).toMatchSnapshot();
})