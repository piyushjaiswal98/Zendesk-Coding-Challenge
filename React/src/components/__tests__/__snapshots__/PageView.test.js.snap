// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches Snapshot 1`] = `
<div
  className="pageview"
  data-testid="PageView-1"
>
  <div
    className="ticketList"
    data-testid="TicketList-1"
  />
  <div
    className="page-class"
  />
</div>
`;
